hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -378,7 +379,10 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	})
 
 	// Validate config input.
-	config, err := r.getConfig(ctx, nodePool)
+	// 3 expectedCoreConfigResources: fips, ssh and haproxy.
+	// TODO (alberto): consider moving the expectedCoreConfigResources check
+	// into the token Secret controller so we don't block Machine infra creation on this.
+	config, err := r.getConfig(ctx, nodePool, 3)
```

Comment:
ibm will have more core resources

Reply:
(as long as check doesn't fail if there are more than 3 this will work).

Reply:
It be nice if that could be "configurable" at least on a per provider basis. (aka IBM Cloud, AWS, etc)